controllers/snapshot/snapshot_suite_test.go

```diff
@@ -99,6 +100,13 @@ var _ = BeforeSuite(func() {
 	k8sClient = k8sManager.GetClient()
 	go func() {
 		defer GinkgoRecover()
+		Expect(cache.SetupIntegrationTestScenarioCache(k8sManager)).To(Succeed())
```

Comment:
I think we want to prevent setting up cache for unit tests because that causes latency issues and works unreliably in gitHub actions (see [STONEINTG-459](https://issues.redhat.com/browse/STONEINTG-459)).

Reply:
It's added because I always met  below issue for the test case "ensures Release created successfully" although the mocked context is added in the case.
-  [FAILED] Unexpected error:
      <*errors.errorString | 0xc00085c6b0>: 
      Index with name field:spec.snapshot does not exist
      {
          s: "Index with name field:spec.snapshot does not exist",
      }
  occurred

Reply:
If we prevent setting up cache, I'll not add the test case in the PR.

Reply:
Just took a look at what's missing and I think that the `Release` resources would also need to be mocked because of [this call](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L401C40-L401C40), so adding something like this to the mocked resources would make it work without cache:
```
       {
		ContextKey: loader.ReleaseContextKey,
		Resource:   &releasev1alpha1.Release{},
	},
```

Reply:
If the "ensures Release created successfully" test case works with the added mocked Release resource I mentioned above, I think it would be good to have it.

Reply:
Yes. It works with the Release resource added in the mocked context as you suggested. Thanks!